en/docs/develop/mi-for-vscode/best-practices.md

```diff
@@ -0,0 +1,234 @@
+## Introduction
+
+This guide provides best practices for designing, developing, deploying, and maintaining integration solutions with WSO2 Micro Integrator (MI). It is intended for integration developers, architects, and DevOps engineers aiming to build scalable, secure, and maintainable solutions.
+
+## Development Best Practices
+
+### General Development Guidelines
+
+- Always use the latest version of the WSO2 Micro Integrator extension for VS Code. It is recommended to update the extension frequently to get the latest features and bug fixes.
+
+- Use the UI provided by the WSO2 Micro Integrator extension for VS Code to create and manage your integration projects. This ensures that your projects are set up correctly and follow best practices. You can use manual editing only for advanced configurations not supported by the UI.
+
+### Creating Projects
+
+- Create a separate project for each microservice or integration flow. This approach enhances organization, streamlines artifact management, and facilitates easier maintenance and updates of individual integrations.
```

Comment:
```suggestion
- Create a separate project for each microservice or integration usecase. This approach enhances organization, streamlines artifact management, and facilitates easier maintenance and updates of individual integrations.
```

Reply:
resolved in [bc39d38](https://github.com/wso2/docs-mi/pull/1789/commits/bc39d38f53345b6fcde9f701464af15698becdc5)